File: api/src/main/java/org/zenika/api/client/ArenaApi.java
package org.zenika.api.client;

import feign.Headers;
import feign.Param;
import feign.RequestLine;
import org.zenika.api.entity.BattleEntity;
import org.zenika.core.BattleStatus;
import org.zenika.core.Pokemon;

import java.util.List;

public interface ArenaApi {

    @RequestLine("POST /fight")
    @Headers("Content-Type: application/json")
    BattleEntity fight(List<Pokemon> pokemons);

    @RequestLine("GET /fight/{id}")
    BattleEntity get(@Param("id") String id);

    @RequestLine("GET /fight/{id}/status")
    BattleStatus getStatus(@Param("id") String id);

}

File: api/src/main/kotlin/org/zenika/api/client/ArenaApi.kt
package org.zenika.api.client

import feign.Headers
import feign.Param
import feign.RequestLine
import org.zenika.api.entity.BattleEntity
import org.zenika.core.Pokemon

interface ArenaApi {

    @RequestLine("POST /fight")
    @Headers("Content-Type: application/json")
    fun fight(pokemons: List<Pokemon>): BattleEntity

    @RequestLine("GET /fight/{id}")
    fun get(@Param("id") id: String): BattleEntity

    @RequestLine("GET /fight/{id}/status")
    fun getStatus(@Param("id") id: String): BattleEntity
}
